yt_dlp/extractor/yle_areena.py

```diff
@@ -114,8 +114,12 @@ def _real_extract(self, url):
                 'ie_key': KalturaIE.ie_key(),
             }
         else:
+            try:
+                ongoing = video_data['data']['ongoing_ondemand']
+            except KeyError:
+                ongoing = video_data['data']['ongoing_event']
```

Comment:
I'd do this, and then also set `is_live = False` before the `if is_podcast:` line up above, as well as returning `'is_live': is_live` in the info dict below
```suggestion
            if traverse_obj(video_data, ('data', 'ongoing_ondemand', 'manifest_url', {url_or_none})):
                manifest_url = video_data['data']['ongoing_ondemand']['manifest_url']
            elif traverse_obj(video_data, ('data', 'ongoing_event', 'manifest_url', {url_or_none})):
                manifest_url = video_data['data']['ongoing_event']['manifest_url']
                is_live = True
            else:
                raise ExtractorError('Unable to extract manifest URL')
```

really the whole extractor could use a good refactoring (e.g. we don't access anything outside of the `'data'` branch in the `video_data` JSON, etc), but that's not necessary now